lib/stringReplaceMiddleware.js

```diff
@@ -3,14 +3,19 @@ const stringReplaceStream = require('./stringReplaceStream');
 
 module.exports = (replacements) => (req, res, next) => {
     hijackResponse(res, function (err, res) {
-        if (err) {
-            res.unhijack(); // Make the original res object work again
-            return next(err);
+        const contentType = res.get('content-type') || "";
+        if (contentType.startsWith('text')) {
```

Comment:
will not work for `application/json` contentType
`options.contentTypeFilterRegexp.test(contentType)`
With a default value for `options.contentTypeFilterRegexp`
`options.contentTypeFilterRegexp = /^text\/|^application\/json$|^application\/xml$/` maybe

Reply:
I added the regex provided as the defauit value.